src/components/NavBar.jsx

```diff
@@ -1,6 +1,31 @@
 import { Link } from "react-router-dom";
+import { useContext } from "react";
+import { UserContext } from "../App";
 
-export function NavBar() {
+export function NavBar({ setUser }) {
+  const user = useContext(UserContext);
```

Comment:
By using `useContext` you should technically not need to pass the `setUser` function down via props, you should be able to de-structure it on line 6, check this section of the notes: [https://notes.northcoders.com/courses/js-front-end/react-context#usecontext](https://notes.northcoders.com/courses/js-front-end/react-context#usecontext )